### DISCUSSION TITLE
Adding missing plugin dependencies to the Windows build

### DISCUSSION THREAD

**Discussion by tridian-tn** [zim-desktop-wiki/zim-desktop-wiki#2563] [Development]
I was looking through the recent commits and saw the PR that fixes https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/2012 (relevant changes: https://github.com/zim-desktop-wiki/zim-desktop-wiki/pull/2386/files), fixing the Windows build so that the Linkmap plugin works. It's been a longstanding bugbear of mine that the Spell Checker plugin doesn't work on the Windows build, and although I know I can run Zim from source with spell-checking support, most of the time I like the stability (and discouragement to tinker) of running the official build.

To that end, and apologies for a question coming from unfamiliarity with the Zim build process, is the solution to the issue above a way forward for other missing plugin dependencies in the Windows build e.g. gtkspell or gtkspellcheck for the Spell Checker plugin, seqdiag for Insert Sequence Diagram? i.e. any dependency that is a python library that can be installed with pip

If so, I'm happy to have a go at learning how it works and submitting a PR

Many thanks

**Comment by introt** [comment]
Yes, missing plugin dependencies are fixed by adding those dependencies to the build file. Dependencies that are available on the Python package index can be installed using `pip`, others available from the mingw repos can be installed using `pacman`, and the rest can be included by building them according to their separate instructions.

This is trivial in most cases, but we do require testing prior to submitting a PR which further requires the submitter to have a fully functional build chain on their machine. Furthermore, the size of the build increases with every included package, so including some dependencies out of the box can become infeasible. I'm actually pretty surprised that numpy made the cut, as it may almost double the size of our executable unless I'm completely mistaken.

**Reply by tridian-tn** [reply]
Ok great thanks. So I could in theory approach each dependency in turn and submit results of a before-and-after executable size for discussion?

**Reply by introt** [reply]
Yes. At some point it might make sense to do two builds - one with a minimal set of dependencies, and a full-fat one with all the bells and whistles. I'd imagine there is a way to modify the environment variables to expose non-bundled libraries, but I'm not very familiar with MSYS2 or Windows in general.

**Comment by jaap-karssenberg** [comment]
Keep in mind though, that for spell checking to work, you also need the
libraries. We could of course package a few languages, but it will never be
exhaustive.

--- Jaap

**Reply by tridian-tn** [reply]
Very good point, thanks. Plenty for me to get to grips with then!

**Reply by introt** [reply]
Do check out https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/2047#discussioncomment-8484736 - they've demonstrated that it's possible to install stuff - gtkspell, specifically - on top of the portable Windows install.

**Reply by tridian-tn** [reply]
That sounds promising. I'll give that a go and see if I get the same results. If there was then an updated wiki page (https://github.com/jaap-karssenberg/zim-wiki/wiki/Install-spell-checking-on-windows) with instructions that worked for the Windows/installed version then that could (a) take care of worries over packaging concerns in the main installer, and (b) maybe at some point give the opportunity for a simple add-on installer that installs the spell-check dependencies over the top of an existing installation.

While I'm here, for future reference for consideration for installers, in an example of a hunspell directory I found on my machine 18 languages came to 55MB. Zipping it (like an installer would) took it down to 14MB.
